src/core/integrate.cpp

```diff
@@ -241,37 +260,29 @@ void integrate_vv(int n_steps, int reuse_forces) {
 
 #endif
 
-    /* Integration Steps: Step 1 and 2 of Velocity Verlet scheme:
-       v(t+0.5*dt) = v(t) + 0.5*dt * a(t)
-       p(t + dt)   = p(t) + dt * v(t+0.5*dt)
-       NOTE: Depending on the integration method Step 1 and Step 2
-       cannot be combined for the translation.
-    */
-    if (integ_switch == INTEG_METHOD_NPT_ISO) {
-      propagate_vel(particles);
-      propagate_pos(particles);
-
-      /* Propagate time: t = t+dt */
-      sim_time += time_step;
-    } else if (integ_switch == INTEG_METHOD_STEEPEST_DESCENT) {
+    if (integ_switch == INTEG_METHOD_STEEPEST_DESCENT) {
       if (steepest_descent_step(particles))
         break;
     } else {
-      propagate_vel_pos(particles);
-
-      /* Propagate time: t = t+dt */
-      sim_time += time_step;
+      /* Integration Steps: Step 1 and 2 of Velocity Verlet scheme:
+         v(t+0.5*dt) = v(t) + 0.5*dt * a(t)
+         p(t + dt)   = p(t) + dt * v(t+0.5*dt)
+         NOTE: Depending on the integration method Step 1 and Step 2
+         cannot be combined for the translation.
+      */
+      velocity_verlet_step_1_and_2(particles);
     }
 
+    /* Propagate time: t = t+dt */
+    sim_time += time_step;
```

Comment:
Energy minimization should not propagate time.

Reply:
This is fundamental thing here, time and time steps belong to a specific integrator. There are integrators that have no concept of time, like steepest descent.